gomatic/gocd/config_repos.py

```diff
@@ -43,28 +66,35 @@ def __eq__(self, other):
 
 
 class ConfigRepos(CommonEqualityMixin):
+
     def __init__(self, element, configurator):
         self.element = element
         self.__configurator = configurator
 
     @property
     def config_repo(self):
-        return [ConfigRepo(e) for e in self.element.findall('config-repo')]
+        return [ConfigRepo(e, self.__configurator.server_version) for e in self.element.findall('config-repo')]
 
     def make_empty(self):
         PossiblyMissingElement(self.element).remove_all_children()
 
-    def ensure_config_repo(self, url, plugin, cvs='git', configuration=None):
+    def ensure_config_repo(self, url, plugin, cvs='git', configuration=None, repo_id=None):
         configuration_xml_string = ""
         if configuration:
             configuration_xml_string = '<configuration>{}</configuration>'.format(
                 "".join("<property><key>{0}</key><value>{1}</value></property>".format(k, v) for k, v in
                         configuration.items()))
-
-        element = ET.fromstring('<config-repo plugin="{0}"><{2} url="{1}" />{3}</config-repo>'.format(
-            plugin, url, cvs, configuration_xml_string))
-
-        config_repo_element = ConfigRepo(element)
+        if has_id(self.__configurator.server_version):
+            if not repo_id:
+                repo_id = url.replace(':', '').replace('/', '-')
```

Comment:
maybe a guid instead? thats what gocd does if im not mistaken

Reply:
That was what I thought as well, but if there is a way to make the server generate an ID I can't find it. Trying to manually add them it just shouts about the lack of id. Will make that change though.